pallets/liquid-staking/src/types.rs

```diff
@@ -88,59 +86,115 @@ pub struct StakingBondCall<T: Config> {
 
 /// Relaychain staking.bond_extra call arguments
 #[derive(Clone, Encode, Decode, RuntimeDebug)]
-pub struct StakingBondExtraCall<Balance> {
-    /// [pallet index, call index]
-    pub call_index: [u8; 2],
+pub struct StakingBondExtraCall<T: Config> {
     /// Rebond amount
     #[codec(compact)]
-    pub value: Balance,
+    pub value: BalanceOf<T>,
 }
 
 /// Relaychain staking.unbond call arguments
 #[derive(Clone, Encode, Decode, RuntimeDebug)]
-pub struct StakingUnbondCall<Balance> {
-    /// [pallet index, call index]
-    pub call_index: [u8; 2],
+pub struct StakingUnbondCall<T: Config> {
     /// Unbond amount
     #[codec(compact)]
-    pub value: Balance,
+    pub value: BalanceOf<T>,
 }
 
 /// Relaychain staking.rebond call arguments
 #[derive(Clone, Encode, Decode, RuntimeDebug)]
-pub struct StakingRebondCall<Balance> {
-    /// [pallet index, call index]
-    pub call_index: [u8; 2],
+pub struct StakingRebondCall<T: Config> {
     /// Rebond amount
     #[codec(compact)]
-    pub value: Balance,
+    pub value: BalanceOf<T>,
 }
 
 /// Relaychain staking.withdraw_unbonded call arguments
 #[derive(Clone, Encode, Decode, RuntimeDebug)]
 pub struct StakingWithdrawUnbondedCall {
-    /// [pallet index, call index]
-    pub call_index: [u8; 2],
     /// Withdraw amount
     pub num_slashing_spans: u32,
 }
 
 /// Relaychain staking.nominate call arguments
 #[derive(Clone, Encode, Decode, RuntimeDebug)]
 pub struct StakingNominateCall<T: Config> {
-    /// [pallet index, call index]
-    pub call_index: [u8; 2],
     /// List of nominate `targets`
     pub targets: Vec<<T::Lookup as StaticLookup>::Source>,
 }
 
 /// Relaychain staking.payout_stakers call arguments
 #[derive(Clone, Encode, Decode, RuntimeDebug)]
-pub struct StakingPayoutStakersCall<AccountId> {
-    /// [pallet index, call index]
-    pub call_index: [u8; 2],
+pub struct StakingPayoutStakersCall<T: Config> {
     /// Stash account of validator
-    pub validator_stash: AccountId,
+    pub validator_stash: T::AccountId,
     /// EraIndex
     pub era: u32,
 }
+
+#[derive(Encode, Decode, RuntimeDebug)]
+pub enum StakingCall<T: Config> {
+    #[codec(index = 0)]
+    Bond(StakingBondCall<T>),
+    #[codec(index = 1)]
+    BondExtra(StakingBondExtraCall<T>),
+    #[codec(index = 2)]
+    Unbond(StakingUnbondCall<T>),
+    #[codec(index = 19)]
+    Rebond(StakingRebondCall<T>),
+    #[codec(index = 3)]
+    WithdrawUnbonded(StakingWithdrawUnbondedCall),
+    #[codec(index = 5)]
+    Nominate(StakingNominateCall<T>),
+    #[codec(index = 18)]
+    PayoutStakers(StakingPayoutStakersCall<T>),
+}
```

Comment:
can we make this increasing sequence according to the index?

Reply:
sure